File: snippets/avacat.js
function crc32 (str) {
  const crcTable = [];
  let c

  for (let i = 0; i < 256; i++) {
      c = i
      for (let j = 0; j < 8; j++){
          c = ((c & 1) ? (0xEDB88320 ^ (c >>> 1)) : (c >>> 1))
      }
      crcTable[i] = c;
  }

  let crc = 0 ^ (-1);

  for (let i = 0; i < str.length; i++) {
      crc = (crc >>> 8) ^ crcTable[(crc ^ str.charCodeAt(i)) & 0xFF]
  }

  return (crc ^ (-1)) >>> 0
}

function avacat (name, size, num = 255) {
  const i = crc32(name) % num
  return `https://shantichat.github.io/avacats/${size}x${size}/${i}.jpg`
}

console.log(avacat('alice@example.com', 80))
// https://shantichat.github.io/avacats/80/171.jpg

console.log(avacat('bob@example.com', 120))
// https://shantichat.github.io/avacats/120/222.jpg

File: snippets/avacat.go
package main

import (
	"fmt"
	"hash/crc32"
)

func main() {
	fmt.Println(Avacat("alice@example.com", 80)) // https://shantichat.github.io/avacats80x80/171.jpg
	fmt.Println(Avacat("bob@example.com", 120))  // https://shantichat.github.io/avacats120x120/222.jpg
}

func Avacat(name string, size int) string {
	const root = "https://shantichat.github.io/avacats"
	const maxNum = 255
	i := crc32.ChecksumIEEE([]byte(name)) % maxNum
	return fmt.Sprintf("%s/%dx%d/%d.jpg", root, size, size, i)
}
